calico_containers/pycalico/util.py

```diff
@@ -104,211 +108,434 @@ def get_hostname():
         return socket.gethostname()
 
 
-def validate_port_str(port_str):
+def return_bool(func, *args, **kwargs):
     """
-    Checks whether the command line word specifying a set of ports is valid.
+    Simple function to catch exceptions and return a Bool
+
+    :param func: The function being run
+    :param *args: All positional args being passed to func
+    :param **kwargs: All key-value args being passed to func
+    :return: True if function succeeds, False if exception is raised
+    :rtype: bool
     """
-    return validate_ports(port_str.split(","))
+    try:
+        func(*args, **kwargs)
+    except Exception:  # pylint: disable=broad-except
+        return False
+    else:
+        return True
 
 
-def validate_ports(port_list):
+def validate_asn(asn):
     """
-    Checks whether a list of ports are within range of 0 and 65535.
-    The port list must include a number or a number range.
+    DEPRECATED (use verify_asn)
 
-    A valid number range must be two numbers delimited by a colon with the
-    second number higher than the first. Both numbers must be within range.
-    If a number range is invalid, the function will return False.
+    Validate the format of a 2-byte or 4-byte autonomous system number
 
-    :param port_list:
-    :return: a Boolean: True if in range, False if not in range
+    :param asn: User input of AS number
+    :type asn: str
+    :return: True if valid format, False if invalid format
+    :rtype: bool
     """
-    in_range = True
-    for port in port_list:
-        if ":" in str(port):
-            ports = port.split(":")
-            in_range = (len(ports) == 2) and (int(ports[0]) < int(ports[1])) \
-                       and validate_ports(ports)
-        else:
-            try:
-                in_range = 0 <= int(port) < 65536
-            except ValueError:
-                in_range = False
-        if not in_range:
-            break
+    return return_bool(validate_asn, asn)
```

Comment:
Oh! I didn't even notice there's no unit tests for validate_asn. I'll fix that as well.